Flaky-test: ReplicatorTest.testResumptionAfterBacklogRelaxed

### Search before reporting

- [x] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/18018071040/job/51268675975?pr=24784#step:11:1094

### Exception stacktrace

Full logs at https://gist.github.com/lhotari/13060eeb18d3bd0bd0a694ba27f0c35d

```text
Error:  Tests run: 237, Failures: 1, Errors: 0, Skipped: 236, Time elapsed: 51.16 s <<< FAILURE! -- in org.apache.pulsar.broker.service.ReplicatorTest
  Error:  org.apache.pulsar.broker.service.ReplicatorTest.testResumptionAfterBacklogRelaxed -- Time elapsed: 11.82 s <<< FAILURE!
  java.lang.AssertionError: 
  
  Expecting any element of:
    [ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.memory.committed, description=Measure of memory committed, unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'non-profiled nmethods'", type="non_heap"}, value=29425664, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'non-nmethods'", type="non_heap"}, value=3538944, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="ZHeap", type="heap"}, value=1063256064, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="Compressed Class Space", type="non_heap"}, value=20840448, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="Metaspace", type="non_heap"}, value=177012736, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'profiled nmethods'", type="non_heap"}, value=46989312, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.memory.usage_after_last_gc, description=Measure of memory used after the most recent garbage collection event on this pool, unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="ZHeap", type="heap"}, value=444596224, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.gc.duration, description=Duration of JVM garbage collection actions, unit=s, type=HISTOGRAM, data=ImmutableHistogramData{aggregationTemporality=CUMULATIVE, points=[ImmutableHistogramPointData{getStartEpochNanos=1758829443697133096, getEpochNanos=1758829458473441148, getAttributes={action="end of GC pause", gc="ZGC Pauses"}, getSum=0.0, getCount=30, hasMin=true, getMin=0.0, hasMax=true, getMax=0.0, getBoundaries=[], getCounts=[30], getExemplars=[]}, ImmutableHistogramPointData{getStartEpochNanos=1758829443697133096, getEpochNanos=1758829458473441148, getAttributes={action="end of GC cycle", gc="ZGC Cycles"}, getSum=4.882, getCount=10, hasMin=true, getMin=0.311, hasMax=true, getMax=0.884, getBoundaries=[], getCounts=[10], getExemplars=[]}]}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.system.cpu.load_1m, description=Average CPU load of the whole system for the last minute, unit={run_queue_item}, type=DOUBLE_GAUGE, data=ImmutableGaugeData{points=[ImmutableDoublePointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={}, value=2.7880859375, exemplars=[]}]}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.memory.limit, description=Measure of max obtainable memory, unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'non-profiled nmethods'", type="non_heap"}, value=122912768, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'non-nmethods'", type="non_heap"}, value=5836800, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="ZHeap", type="heap"}, value=1363148800, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="Compressed Class Space", type="non_heap"}, value=1073741824, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'profiled nmethods'", type="non_heap"}, value=122908672, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.system.cpu.utilization, description=Recent cpu utilization for the whole system, unit=1, type=DOUBLE_GAUGE, data=ImmutableGaugeData{points=[ImmutableDoublePointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={}, value=0.33819702881152464, exemplars=[]}]}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.classes.loaded, description=Number of classes loaded since JVM start, unit={class}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={}, value=32095, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.buffer.usage, description=Memory that the Java virtual machine is using for this buffer pool, unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="mapped"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="direct"}, value=6468041, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="mapped - 'non-volatile memory'"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.cpu.utilization, description=Recent cpu utilization for the process, unit=1, type=DOUBLE_GAUGE, data=ImmutableGaugeData{points=[ImmutableDoublePointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={}, value=0.025, exemplars=[]}]}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.threads.count, description=Number of executing threads, unit={thread}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={daemon=false}, value=440, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={daemon=true}, value=138, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.memory.usage, description=Measure of memory used, unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'non-profiled nmethods'", type="non_heap"}, value=28860416, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'non-nmethods'", type="non_heap"}, value=3182464, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="ZHeap", type="heap"}, value=467664896, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="Compressed Class Space", type="non_heap"}, value=19420488, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="Metaspace", type="non_heap"}, value=174222104, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'profiled nmethods'", type="non_heap"}, value=41147904, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.classes.current_loaded, description=Number of classes currently loaded, unit={class}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={}, value=28161, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.buffer.count, description=The number of buffers in the pool, unit={buffers}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="mapped"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="direct"}, value=269, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="mapped - 'non-volatile memory'"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.buffer.limit, description=Total capacity of the buffers in this pool, unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="mapped"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="direct"}, value=6468040, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="mapped - 'non-volatile memory'"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.classes.unloaded, description=Number of classes unloaded since JVM start, unit={class}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={}, value=3934, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-telemetry-java8, version=1.33.6-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.memory.init, description=Measure of initial memory requested, unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'non-profiled nmethods'", type="non_heap"}, value=2555904, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'non-nmethods'", type="non_heap"}, value=2555904, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="ZHeap", type="heap"}, value=8388608, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="Compressed Class Space", type="non_heap"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="Metaspace", type="non_heap"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pool="CodeHeap 'profiled nmethods'", type="non_heap"}, value=2555904, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.topic.storage.logical.size, description=The storage size of the messages in this topic, excluding storage used by replicas., unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e"}, value=70, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.consumer.message.ack.count, description=The total number of message acknowledgments received from this consumer., unit={ack}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.consumer.id=0, pulsar.consumer.name="t6QeF", pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.subscription.name="__system_reader-reader-70e01324e6", pulsar.subscription.type="Exclusive", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.managed_ledger.mark_delete.count, description=The total number of mark delete operations for this ledger., unit={operation}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.name="pulsar/ns1/persistent/%25s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.namespace="pulsar/ns1"}, value=1, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.replication.message.outgoing.count, description=The total number of messages sent to the remote cluster through this replicator., unit={message}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.replication.remote.cluster.name="r2", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e"}, value=2, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.replication.remote.cluster.name="r2", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.request.schema_registry.duration, description=The duration of Schema Registry requests., unit=s, type=HISTOGRAM, data=ImmutableHistogramData{aggregationTemporality=CUMULATIVE, points=[ImmutableHistogramPointData{getStartEpochNanos=1758829443697133096, getEpochNanos=1758829458473441148, getAttributes={pulsar.namespace="pulsar/ns1", pulsar.schema_registry.request="put", pulsar.schema_registry.response="success"}, getSum=0.069, getCount=1, hasMin=true, getMin=0.069, hasMax=true, getMax=0.069, getBoundaries=[0.0, 5.0, 10.0, 25.0, 50.0, 75.0, 100.0, 250.0, 500.0, 750.0, 1000.0, 2500.0, 5000.0, 7500.0, 10000.0], getCounts=[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], getExemplars=[]}, ImmutableHistogramPointData{getStartEpochNanos=1758829443697133096, getEpochNanos=1758829458473441148, getAttributes={pulsar.namespace="pulsar/ns1", pulsar.schema_registry.request="get", pulsar.schema_registry.response="success"}, getSum=0.028, getCount=5, hasMin=true, getMin=0.0, hasMax=true, getMax=0.014, getBoundaries=[0.0, 5.0, 10.0, 25.0, 50.0, 75.0, 100.0, 250.0, 500.0, 750.0, 1000.0, 2500.0, 5000.0, 7500.0, 10000.0], getCounts=[2, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], getExemplars=[]}]}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.managed_ledger.message_range.count, description=The number of non-contiguous deleted messages ranges., unit={range}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.name="pulsar/ns1/persistent/%25s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__compaction", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__system_reader-reader-70e01324e6", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.managed_ledger.cursor.outgoing.size, description=The total amount of data written to the ledger., unit={By}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.name="pulsar/ns1/persistent/%25s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.namespace="pulsar/ns1"}, value=34, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__compaction", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__system_reader-reader-70e01324e6", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.topic.message.incoming.count, description=The total number of messages received for this topic., unit={message}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e"}, value=2, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.managed_ledger.cache.entry.count, description=The number of entries in the entry cache., unit={entry}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cache.entry.status="active"}, value=1, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cache.entry.status="inserted"}, value=2, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cache.entry.status="evicted"}, value=1, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.topic.storage.offloaded.size, description=The total amount of the data in this topic offloaded to the tiered storage., unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.connection.create.operation.count, description=The number of connection create operations., unit={operation}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.connection.create.operation.status="failure"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.connection.create.operation.status="success"}, value=8, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.consumer.message.unack.count, description=The total number of messages unacknowledged by this consumer., unit={message}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.consumer.id=0, pulsar.consumer.name="t6QeF", pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.subscription.name="__system_reader-reader-70e01324e6", pulsar.subscription.type="Exclusive", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.topic.transaction.count, description=The number of transactions on this topic., unit={transaction}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.transaction.status="aborted"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events", pulsar.transaction.status="aborted"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.transaction.status="committed"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events", pulsar.transaction.status="committed"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.transaction.status="active"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events", pulsar.transaction.status="active"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.managed_ledger.persist.mds.operation.count, description=The number of acknowledgment operations in the metadata store., unit={operation}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__system_reader-reader-70e01324e6", pulsar.managed_ledger.cursor.operation.status="failure", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__system_reader-reader-70e01324e6", pulsar.managed_ledger.cursor.operation.status="success", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.cursor.operation.status="failure", pulsar.managed_ledger.name="pulsar/ns1/persistent/%25s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.cursor.operation.status="success", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__compaction", pulsar.managed_ledger.cursor.operation.status="success", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.cursor.operation.status="failure", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.cursor.operation.status="success", pulsar.managed_ledger.name="pulsar/ns1/persistent/%25s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__compaction", pulsar.managed_ledger.cursor.operation.status="failure", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.consumer.message.outgoing.size, description=The total number of messages bytes dispatched to this consumer., unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.consumer.id=0, pulsar.consumer.name="t6QeF", pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.subscription.name="__system_reader-reader-70e01324e6", pulsar.subscription.type="Exclusive", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.topic.consumer.count, description=The number of active consumers of the topic connected to this broker., unit={consumer}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=1, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.topic.subscription.count, description=The number of Pulsar subscriptions of the topic served by this broker., unit={subscription}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=2, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.topic.publish.rate.limit.count, description=The number of times the publish rate limit is triggered., unit={event}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.topic.message.incoming.size, description=The total number of messages bytes received for this topic., unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/%s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e"}, value=70, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.consumer.blocked, description=Indicates whether the consumer is currently blocked due to unacknowledged messages., unit=1, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.consumer.id=0, pulsar.consumer.name="t6QeF", pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.subscription.name="__system_reader-reader-70e01324e6", pulsar.subscription.type="Exclusive", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.topic.load.concurrent.usage, description=The number of pending topic load operations in the broker. When it reaches threshold "maxConcurrentTopicLoadRequest" defined in broker.conf, new requests are rejected., unit={operation}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.request.topic.lookup.concurrent.limit, description=The maximum number of pending lookup operations in the broker. Equal to "maxConcurrentLookupRequest" defined in broker.conf., unit={operation}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={}, value=50000, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.managed_ledger.count, description=The total number of managed ledgers., unit={managed_ledger}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={}, value=2, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.managed_ledger.cursor.incoming.size, description=The total amount of data read from the ledger., unit={By}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.name="pulsar/ns1/persistent/%25s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__compaction", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__system_reader-reader-70e01324e6", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
      ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, 
<----------- cut --------------->
[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.name="pulsar/ns1/persistent/%25s-producer_exception-8c3a0098-cd18-4979-b68b-1e2ecf2bac9e", pulsar.namespace="pulsar/ns1"}, value=17, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__compaction", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="__system_reader-reader-70e01324e6", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.managed_ledger.cursor.name="pulsar.repl.r2", pulsar.managed_ledger.name="pulsar/ns1/persistent/__change_events", pulsar.namespace="pulsar/ns1"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
  	at org.apache.pulsar.broker.service.ReplicatorTest.testResumptionAfterBacklogRelaxed(ReplicatorTest.java:1049)
  	...(10 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
  
  ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.consumer.message.outgoing.count, description=The total number of messages dispatched to this consumer., unit={message}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.consumer.id=0, pulsar.consumer.name="t6QeF", pulsar.domain="persistent", pulsar.namespace="pulsar/ns1", pulsar.subscription.name="__system_reader-reader-70e01324e6", pulsar.subscription.type="Exclusive", pulsar.tenant="pulsar", pulsar.topic="persistent://pulsar/ns1/__change_events"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}}
  error: org.opentest4j.AssertionFailedError: Expected MetricData to have name <pulsar.broker.replication.message.backlog.age> but found <pulsar.broker.consumer.message.outgoing.count>
  	at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.lambda$assertMetricDoubleGaugeValue$22(BrokerOpenTelemetryTestUtil.java:116)
  	at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.assertMetricDoubleGaugeValue(BrokerOpenTelemetryTestUtil.java:115)
  	at org.apache.pulsar.broker.service.ReplicatorTest.testResumptionAfterBacklogRelaxed(ReplicatorTest.java:1049)
  	...(10 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
  
  ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.metadata.store.outgoing.size, description=The total amount of data written to the metadata store, unit={By}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.metadata.store.name="configuration-metadata-store"}, value=10820, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.metadata.store.name="metadata-store"}, value=4970, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}}
  error: org.opentest4j.AssertionFailedError: Expected MetricData to have name <pulsar.broker.replication.message.backlog.age> but found <pulsar.broker.metadata.store.outgoing.size>
  	at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.lambda$assertMetricDoubleGaugeValue$22(BrokerOpenTelemetryTestUtil.java:116)
  	at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.assertMetricDoubleGaugeValue(BrokerOpenTelemetryTestUtil.java:115)
  	at org.apache.pulsar.broker.service.ReplicatorTest.testResumptionAfterBacklogRelaxed(ReplicatorTest.java:1049)
  	...(10 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
  
  ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="r1", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.metadata.store.executor.queue.size, description=The number of batch operations in the metadata store executor queue, unit={operation}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.metadata.store.name="configuration-metadata-store"}, value=1, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1758829443697133096, epochNanos=1758829458473441148, attributes={pulsar.metadata.store.name="metadata-store"}, value=1, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}}
  error: org.opentest4j.AssertionFailedError: Expected MetricData to have name <pulsar.broker.replication.message.backlog.age> but found <pulsar.broker.metadata.store.executor.queue.size>
  	at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.lambda$assertMetricDoubleGaugeValue$22(BrokerOpenTelemetryTestUtil.java:116)
  	at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.assertMetricDoubleGaugeValue(BrokerOpenTelemetryTestUtil.java:115)
  	at org.apache.pulsar.broker.service.ReplicatorTest.testResumptionAfterBacklogRelaxed(ReplicatorTest.java:1049)
  	...(10 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
  	at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.assertMetricDoubleGaugeValue(BrokerOpenTelemetryTestUtil.java:115)
  	at org.apache.pulsar.broker.service.ReplicatorTest.testResumptionAfterBacklogRelaxed(ReplicatorTest.java:1049)
  	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
  	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
  	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
  	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
  	at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ReplicatorTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ReplicatorTest.java
@@ -104,6 +104,7 @@ import org.apache.pulsar.common.policies.data.BacklogQuota;
 import org.apache.pulsar.common.policies.data.BacklogQuota.RetentionPolicy;
 import org.apache.pulsar.common.policies.data.ClusterData;
 import org.apache.pulsar.common.policies.data.PartitionedTopicStats;
+import org.apache.pulsar.common.policies.data.Policies;
 import org.apache.pulsar.common.policies.data.ReplicatorStats;
 import org.apache.pulsar.common.policies.data.SchemaCompatibilityStrategy;
 import org.apache.pulsar.common.policies.data.TopicStats;
@@ -972,6 +973,10 @@ public class ReplicatorTest extends ReplicatorTestBase {
         assertTrue(remoteClusters.contains("r1"));
     }
 
+    @DataProvider(name = "retentionPolicies")
+    public static Object[][] retentionPolicies() {
+        return new Object[][] { { RetentionPolicy.producer_exception }, { RetentionPolicy.producer_request_hold } };
+    }
 
     /**
      * Issue #199
@@ -982,91 +987,79 @@ public class ReplicatorTest extends ReplicatorTestBase {
      * @throws Exception
      */
 
-    @Test(timeOut = 60000, priority = -1)
-    public void testResumptionAfterBacklogRelaxed() throws Exception {
-        List<RetentionPolicy> policies = new ArrayList<>();
-        policies.add(RetentionPolicy.producer_exception);
-        policies.add(RetentionPolicy.producer_request_hold);
+    @Test(timeOut = 60000, priority = -1, dataProvider = "retentionPolicies")
+    public void testResumptionAfterBacklogRelaxed(RetentionPolicy policy) throws Exception {
+        // create a unique namespace for this test case to avoid flakiness
+        String namespace = newUniqueName("pulsar/testResumptionAfterBacklogRelaxed");
+        Policies policies = new Policies();
+        policies.backlog_quota_map = Map.of(BacklogQuota.BacklogQuotaType.destination_storage, BacklogQuota.builder()
+                // Use 1Mb quota by default
+                .limitSize(1 * 1024 * 1024)
+                .retentionPolicy(policy)
+                .build());
+        policies.replication_clusters = Set.of("r1", "r2");
+        admin1.namespaces().createNamespace(namespace, policies);
 
-        for (RetentionPolicy policy : policies) {
-            // Use 1Mb quota by default
-            admin1.namespaces().setBacklogQuota("pulsar/ns1", BacklogQuota.builder()
-                    .limitSize(1 * 1024 * 1024)
-                    .retentionPolicy(policy)
-                    .build());
-            Thread.sleep(200);
+        TopicName dest = TopicName.get("persistent://" + namespace + "/" + policy);
 
-            TopicName dest = TopicName
-                    .get(BrokerTestUtil.newUniqueName("persistent://pulsar/ns1/%s-" + policy));
+        // Producer on r1
+        @Cleanup
+        MessageProducer producer1 = new MessageProducer(url1, dest);
 
-            // Producer on r1
-            @Cleanup
-            MessageProducer producer1 = new MessageProducer(url1, dest);
+        // Consumer on r2
+        @Cleanup
+        MessageConsumer consumer2 = new MessageConsumer(url2, dest);
 
-            // Consumer on r2
-            @Cleanup
-            MessageConsumer consumer2 = new MessageConsumer(url2, dest);
+        // Replicator for r1 -> r2
+        PersistentTopic topic = (PersistentTopic) pulsar1.getBrokerService()
+                .getTopicReference(dest.toString()).get();
+        Replicator replicator = topic.getPersistentReplicator("r2");
 
-            // Replicator for r1 -> r2
-            PersistentTopic topic = (PersistentTopic) pulsar1.getBrokerService()
-                    .getTopicReference(dest.toString()).get();
-            Replicator replicator = topic.getPersistentReplicator("r2");
+        // Produce 1 message in r1. This message will be replicated immediately into r2 and it will become part of
+        // local backlog
+        producer1.produce(1);
 
-            // Produce 1 message in r1. This message will be replicated immediately into r2 and it will become part of
-            // local backlog
-            producer1.produce(1);
+        Awaitility.await().untilAsserted(() -> assertEquals(replicator.computeStats().replicationBacklog, 0));
+        var attributes = Attributes.of(
+                OpenTelemetryAttributes.PULSAR_DOMAIN, dest.getDomain().value(),
+                OpenTelemetryAttributes.PULSAR_TENANT, dest.getTenant(),
+                OpenTelemetryAttributes.PULSAR_NAMESPACE, dest.getNamespace(),
+                OpenTelemetryAttributes.PULSAR_TOPIC, dest.getPartitionedTopicName(),
+                OpenTelemetryAttributes.PULSAR_REPLICATION_REMOTE_CLUSTER_NAME, cluster2
+        );
+        var metrics = metricReader1.collectAllMetrics();
+        assertMetricLongSumValue(metrics, OpenTelemetryReplicatorStats.BACKLOG_COUNTER, attributes, 0);
+        assertMetricDoubleGaugeValue(metrics, OpenTelemetryReplicatorStats.DELAY_GAUGE, attributes, 0.0);
 
-            Thread.sleep(500);
+        // Restrict backlog quota limit to 1 byte to stop replication
+        admin1.namespaces().setBacklogQuota(namespace, BacklogQuota.builder()
+                .limitSize(1)
+                .retentionPolicy(policy)
+                .build());
 
-            // Restrict backlog quota limit to 1 byte to stop replication
-            admin1.namespaces().setBacklogQuota("pulsar/ns1", BacklogQuota.builder()
-                    .limitSize(1)
-                    .retentionPolicy(policy)
-                    .build());
+        Thread.sleep((TIME_TO_CHECK_BACKLOG_QUOTA + 1) * 1000);
 
-            Thread.sleep((TIME_TO_CHECK_BACKLOG_QUOTA + 1) * 1000);
 
-            assertEquals(replicator.computeStats().replicationBacklog, 0);
-            var attributes = Attributes.of(
-                    OpenTelemetryAttributes.PULSAR_DOMAIN, dest.getDomain().value(),
-                    OpenTelemetryAttributes.PULSAR_TENANT, dest.getTenant(),
-                    OpenTelemetryAttributes.PULSAR_NAMESPACE, dest.getNamespace(),
-                    OpenTelemetryAttributes.PULSAR_TOPIC, dest.getPartitionedTopicName(),
-                    OpenTelemetryAttributes.PULSAR_REPLICATION_REMOTE_CLUSTER_NAME, cluster2
-            );
-            var metrics = metricReader1.collectAllMetrics();
-            assertMetricLongSumValue(metrics, OpenTelemetryReplicatorStats.BACKLOG_COUNTER, attributes, 0);
-            assertMetricDoubleGaugeValue(metrics, OpenTelemetryReplicatorStats.DELAY_GAUGE, attributes, 0.0);
+        // Next message will not be replicated, because r2 has reached the quota
+        producer1.produce(1);
 
-            // Next message will not be replicated, because r2 has reached the quota
-            producer1.produce(1);
+        Awaitility.await().untilAsserted(() -> assertEquals(replicator.computeStats().replicationBacklog, 1));
 
-            Thread.sleep(500);
+        metrics = metricReader1.collectAllMetrics();
+        assertMetricLongSumValue(metrics, OpenTelemetryReplicatorStats.BACKLOG_COUNTER, attributes, 1);
+        assertMetricDoubleGaugeValue(metrics, OpenTelemetryReplicatorStats.DELAY_GAUGE, attributes,
+                aDouble -> assertThat(aDouble).isPositive());
 
-            assertEquals(replicator.computeStats().replicationBacklog, 1);
-            metrics = metricReader1.collectAllMetrics();
-            assertMetricLongSumValue(metrics, OpenTelemetryReplicatorStats.BACKLOG_COUNTER, attributes, 1);
-            assertMetricDoubleGaugeValue(metrics, OpenTelemetryReplicatorStats.DELAY_GAUGE, attributes,
-                    aDouble -> assertThat(aDouble).isPositive());
+        // Consumer will now drain 1 message and the replication backlog will be cleared
+        consumer2.receive(1);
 
-            // Consumer will now drain 1 message and the replication backlog will be cleared
-            consumer2.receive(1);
+        // Wait until the 2nd message got delivered to consumer
+        consumer2.receive(1);
 
-            // Wait until the 2nd message got delivered to consumer
-            consumer2.receive(1);
-
-            int retry = 10;
-            for (int i = 0; i < retry && replicator.computeStats().replicationBacklog > 0; i++) {
-                if (i != retry - 1) {
-                    Thread.sleep(100);
-                }
-            }
-
-            assertEquals(replicator.computeStats().replicationBacklog, 0);
-            metrics = metricReader1.collectAllMetrics();
-            assertMetricLongSumValue(metrics, OpenTelemetryReplicatorStats.BACKLOG_COUNTER, attributes, 0);
-            assertMetricDoubleGaugeValue(metrics, OpenTelemetryReplicatorStats.DELAY_GAUGE, attributes, 0.0);
-        }
+        Awaitility.await().untilAsserted(() -> assertEquals(replicator.computeStats().replicationBacklog, 0));
+        metrics = metricReader1.collectAllMetrics();
+        assertMetricLongSumValue(metrics, OpenTelemetryReplicatorStats.BACKLOG_COUNTER, attributes, 0);
+        assertMetricDoubleGaugeValue(metrics, OpenTelemetryReplicatorStats.DELAY_GAUGE, attributes, 0.0);
     }
 
     /**

